pkg/installmanager/installmanager.go

```diff
@@ -894,27 +907,37 @@ func (m *InstallManager) generateAssets(cd *hivev1.ClusterDeployment, workerMach
 				m.log.WithError(err).Error("error reading manifest file")
 				return err
 			}
-
-			modifiedManifestBytes, err := patchWorkerMachineSetManifest(manifestBytes, workerMachinePool, vpcID, m.log)
+			if addSecurityGroup {
+				modifiedBytes, err := patchWorkerMachineSetManifest(manifestBytes, workerMachinePool, vpcID, m.log)
```

Comment:
nts: `workerMachinePool` is guaranteed not to be `nil` here because `addSecurityGroup` would be false per L878.